ZenPacks/zenoss/OpenStackInfrastructure/zenpacklib.py

```diff
@@ -300,8 +300,13 @@ def remove(self, app, leaveObjects=False):
                 LOG.info('Removing %s components' % self.id)
                 cat = ICatalogTool(app.zport.dmd)
                 for brain in cat.search(types=self.NEW_COMPONENT_TYPES):
-                    component = brain.getObject()
-                    component.getPrimaryParent()._delObject(component.id)
+                    try:
+                        component = brain.getObject()
+                    except Exception as e:
+                        LOG.error("Trying to index non-existent object %s", e)
```

Comment:
This log message seems to be misleading. We're trying to remove the object here, not index it.